tests/share/models/test_shareobject.py

```diff
@@ -169,3 +170,62 @@ def test_related_objects(self):
         identifier.refresh_from_db()
 
         assert identifier.creative_work == work
+
+
+@pytest.mark.django_db
+class TestDefrankenize:
+
+    def test_must_be_really_sure_about_it(self):
+        work = factories.AbstractCreativeWorkFactory()
+
+        with pytest.raises(ValueError) as e:
+            work.defrankenize()
+        assert e.value.args == ('You have to be really sure about this', )
+
+        with pytest.raises(ValueError) as e:
+            work.defrankenize(True)
+        assert e.value.args == ('You have to be really sure about this', )
+
+        work.defrankenize(im_really_sure_about_this=True)
+
+    def test_resets_name(self):
+        work = factories.AbstractCreativeWorkFactory(title='Not a franken', is_deleted=False)
```

Comment:
This may be sarcastic but the 'Not a fraken' as a title is a little misleading for this test.